### PR TITLE
@internal/@endinternal Doxygen markers

### PR DESCRIPTION
**Pull request by antkmsft**
In our public headers, try marking internal sections of code with the internal tag (http://www.doxygen.nl/manual/commands.html#cmdinternal). Ideally, we'd tell Doxygen to ignore everyting that starts with the underscore. But if that's not possible, this must be one of the ways to exclude internal pieces from documentation. This PR is not neccessarily intended to go in, but at least it is the branch for Daniel to possibly test how useful Doxygen internal tags are. @danieljurek , you don't have to use it if you have other plans in mind, I am just trying to help and reduce the work that you need to do a bit :)

### CODE REVIEW

**Comment by vhvb1989** [comment]
We still to generate Docs with internal for developers working inside az_core :)

File: sdk/core/core/inc/az_context.h [hunk 1]
```diff
@@ -26,16 +26,20 @@ typedef struct az_context az_context;
  */
 struct az_context
 {
+  /// @internal
```
Comment (vhvb1989) [inline sdk/core/core/inc/az_context.h]: looks like this is not a valid Doxygen command
```
 error: unknown command tag name [-Werror,-Wdocumentation-unknown-command]
/// @internal
    ^~~~~~~~~
```

Comment (vhvb1989) [reply sdk/core/core/inc/az_context.h]: internal would only hide the documentation that is written after it
like
```c
/** @brief Some comment that is visible in docs
    @internal Some info not visible in docs due to internal!  applies to all this line
    This is now visible in docs
    @internal for
    multi line internal 
    we use the endofinternal
    @endofinternal
```

Comment (antkmsft) [reply sdk/core/core/inc/az_context.h]: It should be possible to get rid of the -Wdocumentation errors, but I am afraid you're right that internal/endinternal do only apply to documentation comment paragraphs and not whole regions of the file.

**Pull request closed without merging**